relayer/message_relayer.go

```diff
@@ -125,8 +144,79 @@ func (r *messageRelayer) relayMessage(warpMessageInfo *vmtypes.WarpMessageInfo,
 	return nil
 }
 
-// Run collects signatures from nodes by directly querying them via AppRequest, then aggregates the signatures, and constructs the signed warp message.
-func (r *messageRelayer) createSignedMessage(requestID uint32) (*warp.Message, error) {
+func (r *messageRelayer) createSignedMessage() (*warp.Message, error) {
+	r.logger.Info("Fetching aggregate signature from the source chain validators")
+	warpClient, err := warpBackend.NewClient(r.relayer.apiNodeURI, r.relayer.sourceBlockchainID.String())
+	if err != nil {
+		r.logger.Error(
+			"Failed to create warp client",
+			zap.Error(err),
+		)
+		return nil, err
+	}
+	signingSubnetID := r.relayer.sourceSubnetID
+	if r.relayer.sourceSubnetID == constants.PrimaryNetworkID {
+		signingSubnetID, err = r.relayer.pChainClient.ValidatedBy(context.Background(), r.destinationBlockchainID)
+		if err != nil {
+			r.logger.Error(
+				"failed to get validating subnet for destination chain",
+				zap.String("destinationBlockchainID", r.destinationBlockchainID.String()),
+				zap.Error(err),
+			)
+			return nil, err
+		}
```

Comment:
Could this be simplified just to:
```
signingSubnetID = r.destinationBlockchainID
```

I'm guessing not, but I'm not sure why we'd need to make an RPC call to the P-chain to determine the blockchain ID of the signer set.

Reply:
For sending from the primary network, we need to get the destination subnet ID, not the blockchainID, so that we can query the right set of validators. 

The message relayer has the following information:
- destination blockchain ID (from the warp message)
- source subnet ID (from the configuration)
- source blockchainID (also from the configuration)

So we need to get the destination subnet ID via the P-Chain

Reply:
Ah got it, thanks